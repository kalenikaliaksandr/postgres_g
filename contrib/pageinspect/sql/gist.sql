CREATE TABLE test_gist AS SELECT point(i,i) p, i::text t FROM
    generate_series(1,1000) i;
CREATE INDEX test_gist_idx ON test_gist USING gist (p);

\x

SELECT * FROM gist_page_opaque_info(get_raw_page('test_gist_idx', 0));
SELECT * FROM gist_page_opaque_info(get_raw_page('test_gist_idx', 1));
SELECT * FROM gist_page_opaque_info(get_raw_page('test_gist_idx', 2));

SELECT * FROM gist_page_items(get_raw_page('test_gist_idx', 0));
SELECT * FROM gist_page_items(get_raw_page('test_gist_idx', 1)) LIMIT 10;
SELECT * FROM gist_page_items(get_raw_page('test_gist_idx', 2)) LIMIT 10;

DROP TABLE test1;
