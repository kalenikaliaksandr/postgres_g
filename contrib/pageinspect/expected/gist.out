CREATE TABLE test_gist AS SELECT point(i,i) p, i::text t FROM
    generate_series(1,1000) i;
CREATE INDEX test_gist_idx ON test_gist USING gist (p);
\x
SELECT * FROM gist_page_opaque_info(get_raw_page('test_gist_idx', 0));
-[ RECORD 1 ]---------
lsn       | 0/1
nsn       | 0/0
rightlink | 4294967295
flags     | {}

SELECT * FROM gist_page_opaque_info(get_raw_page('test_gist_idx', 1));
-[ RECORD 1 ]---------
lsn       | 0/1
nsn       | 0/0
rightlink | 4294967295
flags     | {leaf}

SELECT * FROM gist_page_opaque_info(get_raw_page('test_gist_idx', 2));
-[ RECORD 1 ]-----
lsn       | 0/1
nsn       | 0/0
rightlink | 1
flags     | {leaf}

SELECT * FROM gist_page_items_bytea(get_raw_page('test_gist_idx', 0));
-[ RECORD 1 ]----------------------------------------------------------------------------------
itemoffset | 1
ctid       | (1,65535)
itemlen    | 40
tuple_data | \x00000100ffff28000000000000c064400000000000c06440000000000000f03f000000000000f03f
-[ RECORD 2 ]----------------------------------------------------------------------------------
itemoffset | 2
ctid       | (2,65535)
itemlen    | 40
tuple_data | \x00000200ffff28000000000000c074400000000000c074400000000000e064400000000000e06440
-[ RECORD 3 ]----------------------------------------------------------------------------------
itemoffset | 3
ctid       | (3,65535)
itemlen    | 40
tuple_data | \x00000300ffff28000000000000207f400000000000207f400000000000d074400000000000d07440
-[ RECORD 4 ]----------------------------------------------------------------------------------
itemoffset | 4
ctid       | (4,65535)
itemlen    | 40
tuple_data | \x00000400ffff28000000000000c084400000000000c084400000000000307f400000000000307f40
-[ RECORD 5 ]----------------------------------------------------------------------------------
itemoffset | 5
ctid       | (5,65535)
itemlen    | 40
tuple_data | \x00000500ffff28000000000000f089400000000000f089400000000000c884400000000000c88440
-[ RECORD 6 ]----------------------------------------------------------------------------------
itemoffset | 6
ctid       | (6,65535)
itemlen    | 40
tuple_data | \x00000600ffff28000000000000208f400000000000208f400000000000f889400000000000f88940
-[ RECORD 7 ]----------------------------------------------------------------------------------
itemoffset | 7
ctid       | (7,65535)
itemlen    | 40
tuple_data | \x00000700ffff28000000000000408f400000000000408f400000000000288f400000000000288f40

SELECT * FROM gist_page_items_bytea(get_raw_page('test_gist_idx', 1)) LIMIT 10;
-[ RECORD 1 ]----------------------------------------------------------------------------------
itemoffset | 1
ctid       | (0,1)
itemlen    | 40
tuple_data | \x0000000001002800000000000000f03f000000000000f03f000000000000f03f000000000000f03f
-[ RECORD 2 ]----------------------------------------------------------------------------------
itemoffset | 2
ctid       | (0,2)
itemlen    | 40
tuple_data | \x00000000020028000000000000000040000000000000004000000000000000400000000000000040
-[ RECORD 3 ]----------------------------------------------------------------------------------
itemoffset | 3
ctid       | (0,3)
itemlen    | 40
tuple_data | \x00000000030028000000000000000840000000000000084000000000000008400000000000000840
-[ RECORD 4 ]----------------------------------------------------------------------------------
itemoffset | 4
ctid       | (0,4)
itemlen    | 40
tuple_data | \x00000000040028000000000000001040000000000000104000000000000010400000000000001040
-[ RECORD 5 ]----------------------------------------------------------------------------------
itemoffset | 5
ctid       | (0,5)
itemlen    | 40
tuple_data | \x00000000050028000000000000001440000000000000144000000000000014400000000000001440
-[ RECORD 6 ]----------------------------------------------------------------------------------
itemoffset | 6
ctid       | (0,6)
itemlen    | 40
tuple_data | \x00000000060028000000000000001840000000000000184000000000000018400000000000001840
-[ RECORD 7 ]----------------------------------------------------------------------------------
itemoffset | 7
ctid       | (0,7)
itemlen    | 40
tuple_data | \x00000000070028000000000000001c400000000000001c400000000000001c400000000000001c40
-[ RECORD 8 ]----------------------------------------------------------------------------------
itemoffset | 8
ctid       | (0,8)
itemlen    | 40
tuple_data | \x00000000080028000000000000002040000000000000204000000000000020400000000000002040
-[ RECORD 9 ]----------------------------------------------------------------------------------
itemoffset | 9
ctid       | (0,9)
itemlen    | 40
tuple_data | \x00000000090028000000000000002240000000000000224000000000000022400000000000002240
-[ RECORD 10 ]---------------------------------------------------------------------------------
itemoffset | 10
ctid       | (0,10)
itemlen    | 40
tuple_data | \x000000000a0028000000000000002440000000000000244000000000000024400000000000002440

SELECT * FROM gist_page_items_bytea(get_raw_page('test_gist_idx', 2)) LIMIT 10;
-[ RECORD 1 ]----------------------------------------------------------------------------------
itemoffset | 1
ctid       | (1,10)
itemlen    | 40
tuple_data | \x000001000a0028000000000000e064400000000000e064400000000000e064400000000000e06440
-[ RECORD 2 ]----------------------------------------------------------------------------------
itemoffset | 2
ctid       | (1,11)
itemlen    | 40
tuple_data | \x000001000b0028000000000000006540000000000000654000000000000065400000000000006540
-[ RECORD 3 ]----------------------------------------------------------------------------------
itemoffset | 3
ctid       | (1,12)
itemlen    | 40
tuple_data | \x000001000c0028000000000000206540000000000020654000000000002065400000000000206540
-[ RECORD 4 ]----------------------------------------------------------------------------------
itemoffset | 4
ctid       | (1,13)
itemlen    | 40
tuple_data | \x000001000d0028000000000000406540000000000040654000000000004065400000000000406540
-[ RECORD 5 ]----------------------------------------------------------------------------------
itemoffset | 5
ctid       | (1,14)
itemlen    | 40
tuple_data | \x000001000e0028000000000000606540000000000060654000000000006065400000000000606540
-[ RECORD 6 ]----------------------------------------------------------------------------------
itemoffset | 6
ctid       | (1,15)
itemlen    | 40
tuple_data | \x000001000f0028000000000000806540000000000080654000000000008065400000000000806540
-[ RECORD 7 ]----------------------------------------------------------------------------------
itemoffset | 7
ctid       | (1,16)
itemlen    | 40
tuple_data | \x00000100100028000000000000a065400000000000a065400000000000a065400000000000a06540
-[ RECORD 8 ]----------------------------------------------------------------------------------
itemoffset | 8
ctid       | (1,17)
itemlen    | 40
tuple_data | \x00000100110028000000000000c065400000000000c065400000000000c065400000000000c06540
-[ RECORD 9 ]----------------------------------------------------------------------------------
itemoffset | 9
ctid       | (1,18)
itemlen    | 40
tuple_data | \x00000100120028000000000000e065400000000000e065400000000000e065400000000000e06540
-[ RECORD 10 ]---------------------------------------------------------------------------------
itemoffset | 10
ctid       | (1,19)
itemlen    | 40
tuple_data | \x00000100130028000000000000006640000000000000664000000000000066400000000000006640

SELECT * FROM gist_page_items('test_gist_idx', 0);
-[ RECORD 1 ]-----------------
itemoffset | 1
ctid       | (1,65535)
itemlen    | 40
keys       | (p)=((166,166))
-[ RECORD 2 ]-----------------
itemoffset | 2
ctid       | (2,65535)
itemlen    | 40
keys       | (p)=((332,332))
-[ RECORD 3 ]-----------------
itemoffset | 3
ctid       | (3,65535)
itemlen    | 40
keys       | (p)=((498,498))
-[ RECORD 4 ]-----------------
itemoffset | 4
ctid       | (4,65535)
itemlen    | 40
keys       | (p)=((664,664))
-[ RECORD 5 ]-----------------
itemoffset | 5
ctid       | (5,65535)
itemlen    | 40
keys       | (p)=((830,830))
-[ RECORD 6 ]-----------------
itemoffset | 6
ctid       | (6,65535)
itemlen    | 40
keys       | (p)=((996,996))
-[ RECORD 7 ]-----------------
itemoffset | 7
ctid       | (7,65535)
itemlen    | 40
keys       | (p)=((1000,1000))

SELECT * FROM gist_page_items('test_gist_idx', 1) LIMIT 10;
-[ RECORD 1 ]-------------
itemoffset | 1
ctid       | (0,1)
itemlen    | 40
keys       | (p)=((1,1))
-[ RECORD 2 ]-------------
itemoffset | 2
ctid       | (0,2)
itemlen    | 40
keys       | (p)=((2,2))
-[ RECORD 3 ]-------------
itemoffset | 3
ctid       | (0,3)
itemlen    | 40
keys       | (p)=((3,3))
-[ RECORD 4 ]-------------
itemoffset | 4
ctid       | (0,4)
itemlen    | 40
keys       | (p)=((4,4))
-[ RECORD 5 ]-------------
itemoffset | 5
ctid       | (0,5)
itemlen    | 40
keys       | (p)=((5,5))
-[ RECORD 6 ]-------------
itemoffset | 6
ctid       | (0,6)
itemlen    | 40
keys       | (p)=((6,6))
-[ RECORD 7 ]-------------
itemoffset | 7
ctid       | (0,7)
itemlen    | 40
keys       | (p)=((7,7))
-[ RECORD 8 ]-------------
itemoffset | 8
ctid       | (0,8)
itemlen    | 40
keys       | (p)=((8,8))
-[ RECORD 9 ]-------------
itemoffset | 9
ctid       | (0,9)
itemlen    | 40
keys       | (p)=((9,9))
-[ RECORD 10 ]------------
itemoffset | 10
ctid       | (0,10)
itemlen    | 40
keys       | (p)=((10,10))

SELECT * FROM gist_page_items('test_gist_idx', 2) LIMIT 10;
-[ RECORD 1 ]---------------
itemoffset | 1
ctid       | (1,10)
itemlen    | 40
keys       | (p)=((167,167))
-[ RECORD 2 ]---------------
itemoffset | 2
ctid       | (1,11)
itemlen    | 40
keys       | (p)=((168,168))
-[ RECORD 3 ]---------------
itemoffset | 3
ctid       | (1,12)
itemlen    | 40
keys       | (p)=((169,169))
-[ RECORD 4 ]---------------
itemoffset | 4
ctid       | (1,13)
itemlen    | 40
keys       | (p)=((170,170))
-[ RECORD 5 ]---------------
itemoffset | 5
ctid       | (1,14)
itemlen    | 40
keys       | (p)=((171,171))
-[ RECORD 6 ]---------------
itemoffset | 6
ctid       | (1,15)
itemlen    | 40
keys       | (p)=((172,172))
-[ RECORD 7 ]---------------
itemoffset | 7
ctid       | (1,16)
itemlen    | 40
keys       | (p)=((173,173))
-[ RECORD 8 ]---------------
itemoffset | 8
ctid       | (1,17)
itemlen    | 40
keys       | (p)=((174,174))
-[ RECORD 9 ]---------------
itemoffset | 9
ctid       | (1,18)
itemlen    | 40
keys       | (p)=((175,175))
-[ RECORD 10 ]--------------
itemoffset | 10
ctid       | (1,19)
itemlen    | 40
keys       | (p)=((176,176))

DROP TABLE test1;
